Source/Core/Core/HW/DSPHLE/UCodes/AESnd.h

```diff
@@ -29,6 +29,9 @@ class AESndUCode final : public UCodeInterface
   void SetUpAccelerator(u16 format, u16 gain);
   void DoMixing();
 
+  inline bool SwapLeftRight() const;
+  inline bool UseNewFlagMasks() const;
+
```

Comment:
`inline` is useless here, you're telling the compiler that these methods may be defined in multiple translation units but not actually providing the implementation in the header so it never will be.

Reply:
Does `inline` in `AESnd.cpp` have meaning, and if so, does using `inline` in `AESnd.cpp` require also using it `AESnd.h`? I know I've done it in the cpp file before and I did it in the header as well, but I'm not 100% confident I understand the rules surrounding it.

Reply:
I'm not actually sure what a conforming C++ compiler/linker does when you use different linkage declarations across multiple declarations/definitions on the same function (MSVC appears to use the one in the declaration rather than the definition), but to clear up a likely misunderstanding here: `inline` does not mean that the function will be inlined (or considered for inlining), it just affects linkage. (See the first two answers [here](https://stackoverflow.com/questions/1759300/when-should-i-write-the-keyword-inline-for-a-function-method) for details.) The way you've used it here doesn't do anything in practice because there's no possible second compilation unit that could contain a definition.

Reply:
Thanks. In particular this was useful:

> Generally, the compiler will be able to do this better than you. However, the compiler doesn't have the option to inline code if it doesn't have the function definition. In maximally optimized code usually all private methods are inlined whether you ask for it or not.

I made a similar mistake in #10826; I'll change it locally there too. (I have a few other ideas for changes I want to make to that now that I've read the ppc 750cl manual further, to better support dcache emulation later on. So I've made it a draft for now.)

Reply:
(We do also have `DOLPHIN_FORCE_INLINE` in `Common/Inline.h`, but I think that's overkill here.)